src/Storages/VirtualColumnUtils.cpp

```diff
@@ -183,23 +183,23 @@ HivePartitioningKeysAndValues parseHivePartitioningKeysAndValues(const String &
     return key_values;
 }
 
-VirtualColumnsDescription getVirtualsForFileLikeStorage(ColumnsDescription & storage_columns, const ContextPtr & context, const std::string & path, std::optional<FormatSettings> format_settings_)
+VirtualColumnsDescription getVirtualsForFileLikeStorage(ColumnsDescription & storage_columns, const ContextPtr & context, const std::string & path, std::optional<FormatSettings> format_settings_, const NamesAndTypesList & partition_columns)
 {
     VirtualColumnsDescription desc;
 
-    auto add_virtual = [&](const NameAndTypePair & pair, bool prefer_virtual_column) /// By using prefer_virtual_column we define whether we will overwrite the storage column with the virtual one
+    auto add_virtual = [&](const NameAndTypePair & pair, bool allow_duplicate)
     {
         const auto & name = pair.getNameInStorage();
         const auto & type = pair.getTypeInStorage();
         if (storage_columns.has(name))
         {
-            if (!prefer_virtual_column)
+            if (!allow_duplicate)
                 return;
 
             if (storage_columns.size() == 1)
                 throw Exception(ErrorCodes::INCORRECT_DATA, "Cannot use hive partitioning for file {}: it contains only partition columns. Disable use_hive_partitioning setting to read/write this file", path);
+
             auto local_type = storage_columns.get(name).type;
-            storage_columns.remove(name);
```

Comment:
Are you sure that  ClickHouse works correctly everywhere with two columns with same name? I mean not only in simple "SELECT", but in queries "SELECT FROM SELECT ... JOIN ... GROUP BY ..."?

Reply:
Nope, nnot at all, I was hoping CI/CD would show me something.

Honestly, this thing is broken right now, and it's not clear how to fix it